modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java

```diff
@@ -596,6 +597,23 @@ public static void waitUntilRunningQueriesCount(SqlQueryProcessor queryProcessor
         Awaitility.await().untilAsserted(() -> assertThat(queryProcessor.runningQueries().size(), matcher));
     }
 
+    /**
+     * Waits until the number of queries in given phase matches the specified matcher.
+     *
+     * @param queryProcessor The query processor to derive list of running queries.
+     * @param phase The {@link ExecutionPhase} of the interest.
+     * @param matcher THe matcher to check the number of running queries.
+     * @throws AssertionError If after waiting the number of running queries still does not match the specified matcher.
+     */
+    public static void waitUntilQueriesInPhaseCount(SqlQueryProcessor queryProcessor, ExecutionPhase phase, Matcher<Integer> matcher) {
+        Awaitility.await().untilAsserted(() -> assertThat(
+                (int) queryProcessor.runningQueries().stream()
+                        .filter(queryInfo -> queryInfo.phase() == phase)
+                        .count(),
+                matcher
+        ));
```

Comment:
You probably know this, but it can be simplified.
```suggestion
        Awaitility.await().until(
                () -> (int) queryProcessor.runningQueries().stream()
                        .filter(queryInfo -> queryInfo.phase() == phase)
                        .count(),
                matcher
        );
```